R_tree.py

```diff
@@ -263,13 +264,73 @@ def split_node(self, node: Block) -> None:
 				else:
 					self.split_node(parent_block)
 			
+	
+	def underflowTreatment(self, level : int) -> bool:
+		if level == 0:
+			return False
+		
+		# UT2 If the level is not the root level and this is the first
+		# call of UnderflowTreatment in the given level during the
+		# deletion of one data rectangle, then
+		if level not in RTree.level_underflow:
+			RTree.level_underflow.add(level)
+			return True
+		return False
+
+
+	def delete_data(self, record: Record):
+		"""
+		Delete data from the R-tree
+		:param record: Record object to delete
+		:return: None
+		"""
+		self.delete(record)
+
+		RTree.level_underflow.clear()
 			
 	def delete(self, record: Record):
 		"""
 		:param record: Record object to delete
 		:return: None
 		"""
-		pass
+		re_insertions = []  # list of reinserted elements
+		re_insert_flag = False
+		record_bounds = BoundingArea(bounds=BoundingArea.find_bounds_of_records([record]), next_block=None)
+
+		# decrease upper bound and lower bound by 0.001 to avoid floating point errors
+		for bound in record_bounds.bounds:
+			bound.upper += 0.001
+			bound.lower -= 0.001
+
+		stack = [self.root]
+		while len(stack) > 0:
+			node = stack.pop()  # pop the last element / index = -1 by default
+			if node.is_leaf:  # node is leaf, so it contains records
+				if record in node.elements:  # check if the record is in the query area
```

Comment:
αυτος ο ελεγχος ειναι περιττος. η block.delete επιστρέφει false αν δεν βρεθει το στοιχειο

Reply:
εδω η node.delete δηλαδη για να ειναι πιο ομορφουλι

Reply:
ποιον ελεγχο λες; το if record in node.elements?